raiden/utils/filters.py

```diff
@@ -103,19 +112,43 @@ def __init__(self, web3: Web3, filter_params: dict):
         self._last_block: int = -1
         self._lock = Semaphore()
 
-    def get_new_entries(self, block_number: BlockSpecification):
+    def _do_get_new_entries(self, from_block: BlockSpecification, to_block: BlockSpecification):
+        filter_params = self.filter_params.copy()
+        filter_params['fromBlock'] = from_block
+        filter_params['toBlock'] = to_block
+
+        log.debug(f'Querying filter from block {from_block} to block {to_block}')
+        result = self.web3.eth.getLogs(filter_params)
+        self._last_block = to_block
+        return result
+
+    def get_new_entries(self, target_block_number: BlockSpecification):
+        latest_block_number = self.web3.eth.blockNumber
         with self._lock:
-            filter_params = self.filter_params.copy()
-            filter_params['fromBlock'] = max(
-                filter_params.get('fromBlock', GENESIS_BLOCK_NUMBER),
+            result = []
+            from_block = max(
+                self.filter_params.get('fromBlock', GENESIS_BLOCK_NUMBER),
                 self._last_block + 1,
             )
-
-            if self.filter_params.get('toBlock') in (None, 'latest', 'pending'):
-                filter_params['toBlock'] = block_number or 'latest'
-
-            result = self.web3.eth.getLogs(filter_params)
-            self._last_block = filter_params.get('toBlock') or block_number
+            to_block = self.filter_params.get('toBlock')
+            if to_block in (None, 'latest', 'pending'):
+                if not isinstance(target_block_number, int):
+                    target_block_number = latest_block_number
+
+            else:
+                target_block_number = to_block
+
+            # Batch the filter queries in ranges of FILTER_MAX_BLOCK_RANGE
+            # to avoid timeout problems
+            while from_block <= target_block_number:
+                to_block = min(
+                    from_block + FILTER_MAX_BLOCK_RANGE,
+                    target_block_number,
+                )
+                result.extend(
+                    self._do_get_new_entries(from_block=from_block, to_block=to_block),
+                )
+                from_block += FILTER_MAX_BLOCK_RANGE
```

Comment:
Shouldn't this be `from_block = to_clock` in order to handle the second case of the `min` in line 146 above? (in other words, the range is not always `FILTER_MAX_BLOCK_RANGE` blocks long.

Reply:
No. If the second case of the min() happens then this is the last iteration and `from_block` will exceed the target block and succesfully close the loop.

If it was `from_block = to_block` this would become an infinite loop due to the `<=` in the `while`